cfme/tests/infrastructure/test_vm_power_control.py

```diff
@@ -403,20 +403,63 @@ def vm(self, rest_api, vm_name):
         assert result.name == vm_name
         return result
 
-    def test_power_off(self, verify_vm_running, vm):
+    def verify_vm_rest_status(self, vm, status):
+        wait_for(lambda: vm.power_state == status,
+            num_sec=300, delay=5, fail_func=vm.reload,
+            message='wait for VM to stop (current state: {})'.format(vm.power_state))
+
+    def check_finish(self, provider, vm, status):
+        logger.info("VM: {} Power State: {} State on provider: {}".format(
+            vm.name, vm.power_state, provider.mgmt.vm_status(vm.name)))
+        return vm.power_state == status
+
+    def test_stop_detail(self, rest_api, provider, verify_vm_running, vm):
         assert "stop" in vm.action
+        self.verify_vm_rest_status(vm, status='on')
         vm.action.stop()
-        wait_for(lambda: vm.power_state == "off", num_sec=300, delay=5, fail_func=vm.reload)
+        wait_for(lambda: self.check_finish(provider, vm, status="off"),
+            num_sec=600, delay=10, fail_func=vm.reload,
+            message='wait for VM to stop (current state: {})'.format(vm.power_state))
 
-    def test_power_on(self, verify_vm_stopped, vm):
+    def test_power_on_detail(self, rest_api, provider, verify_vm_stopped, vm):
         assert "start" in vm.action
+        self.verify_vm_rest_status(vm, status='off')
```

Comment:
Id prefer to see one of these constants used instead: https://github.com/RedHatQE/cfme_tests/blob/master/cfme/infrastructure/virtual_machines.py#L263\

Same for the other tests.